genvm/templates/vault/vault.go

```diff
@@ -37,29 +39,40 @@ func (v *Vault) isOwner(address core.Address) bool {
 	return v.Owner == address
 }
 
-func (v *Vault) Available(lid core.LayerID) uint64 {
+func (v *Vault) Vested(lid core.LayerID) uint64 {
 	if lid.Before(v.VestingStart) {
 		return 0
 	}
 	if !lid.Before(v.VestingEnd) {
 		return v.TotalAmount
 	}
-	incremental := new(big.Int).SetUint64(v.TotalAmount)
-	incremental.Mul(incremental, new(big.Int).SetUint64(uint64(lid.Difference(v.VestingStart))))
-	incremental.Div(incremental, new(big.Int).SetUint64(uint64(v.VestingEnd.Difference(v.VestingStart))))
-	return incremental.Uint64()
+	vested := new(big.Int).SetUint64(v.TotalAmount)
+	vested.Mul(vested, new(big.Int).SetUint64(uint64(lid.Difference(v.VestingStart))))
+	vested.Div(vested, new(big.Int).SetUint64(uint64(v.VestingEnd.Difference(v.VestingStart))))
+	return vested.Uint64()
 }
 
 // Spend transaction.
 func (v *Vault) Spend(host core.Host, to core.Address, amount uint64) error {
 	if !v.isOwner(host.Principal()) {
 		return ErrNotOwner
 	}
-	available := v.Available(host.Layer())
-	if available > v.TotalAmount {
+	vested := v.Vested(host.Layer())
+
+	// sanity checks
+
+	// cannot vest more than initial endowment
+	if vested > v.TotalAmount {
 		panic("wrong math")
 	}
-	if amount > available-v.DrainedSoFar {
```

Comment:
`DrainedSoFar` is not used anymore - could you remove it?

Reply:
good catch. since the balance check now uses the actual account balance, this is no longer necessary.